mypy/build.py

```diff
@@ -310,6 +310,18 @@ def default_lib_path(data_dir: str, pyversion: Tuple[int, int]) -> List[str]:
 PRI_ALL = 99  # include all priorities
 
 
+def import_priority(imp: ImportBase, toplevel_priority: int) -> int:
+    """Compute import priority from an import node."""
+    if not imp.is_top_level:
+        # Inside a function
+        return PRI_LOW
+    if imp.is_mypy_only:
+        # Inside "if MYPY" or "if typing.TYPE_CHECKING"
+        return max(PRI_MED, toplevel_priority)
```

Comment:
What if we'd make the priority of mypy-only imports even lower, since they can never affect the runtime module initialization order? I feel that more priority levels is generally better, as there will be fewer ties when sorting cycles and thus we should get a more stable ordering. So my suggestion is to create a new priority level for mypy-only imports, which would be either between `PRI_LOW` and `PRI_MED` or even lower than `PRI_LOW`.